estimagic/optimization/pounders.py

```diff
@@ -21,52 +20,52 @@ def minimize_pounders(
-             "func values": np.array of value of the objective at the solution
-             "x": np.array of the start values
-             "conv": string indicating the termination reason
-             "sol": list containing: current iterate as int,
-             current value of the objective as float, current value of
-                    the approximated jacobian as float,
-                     infeasability norm as float, step length as float and termination
-                    reason as int.
+        func (callable): Function that takes a 1d NumPy array and returns a 1d NumPy
+            array.
+        x (np.ndarray): Contains the start values of the variables of interest
+        bounds (list or tuple of lists): Contains the bounds for the variable of
+            interest. The first list contains the lower value for each parameter and the
+            upper list the upper value. The object has to contain two elements of which
+            one represents the upper and the other one the lower bound.
+        init_tr (float): Sets the radius for the initial trust region that the optimizer
+            employs. If none the algorithm uses 100 as initial  trust region radius.
+        max_iterations (int): Alternative Stopping criterion. If set the routine will
+            stop after the number of specified iterations or after the step size is
+            sufficiently small. If the variable is set the default criteria will all be
+            ignored.
+        gatol (int): Stop if relative norm of gradient is less than this. If set to
+            False the algorithm will not consider gatol.
+        grtol (int): Stop if norm of gradient is less than this. If set to False the
+            algorithm will not consider grtol.
+        gttol (int): Stop if norm of gradient is reduced by this factor. If set to False
+            the algorithm will not consider grtol.
 
+    Returns:
+        out (dict): Dictionary with the following key-value pairs:
+
+            - `"solution"`: solution vector as `np.ndarray`.
+            - `"func_values"`: `np.ndarray` of value of the objective at the solution.
+            - `"x"`: `np.ndarray` of the start values.
+            - `"conv"`: string indicating the termination reason.
+            - `"sol"`: `list` containing ...
+              - current iterate as integer.
+              - current value of the objective as float
+              - current value of the approximated
+              - jacobian as float
+              - infeasability norm as float
+              - step length as float
+              - termination reason as int.
 
     """
     if sys.platform == "win32":
         raise NotImplementedError("The pounders algorithm is not available on Windows.")
 
-    # we want to get containers for the func verctor and the paras
+    # We want to get containers for the func verctor and the paras.
     size_paras = len(x)
     size_objective = len_out
```

Comment:
I think we introduced the len_out arg only to avoid an additional evaluation of the objective. I however think it does not really make a difference whether we do that before or after the function call. So I would propose to kick the argument ourt and evaluate the objective in the function for consistency reasons. What do  you think?

Reply:
No, I think it should stay in. If we do this evaluation, we want to do it before we call `minimize_pounders`. Otherwise the evaluation is written to the log file.